lib/web_apis/web_apis.es6.js

```diff
@@ -0,0 +1,253 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+require('./browser_api.es6');
+
+foam.CLASS({
+  name: 'WebAPIs',
+  package: 'org.chromium.apis.web',
+  documentation: `WebAPIs is the main model of this project. It has a
+    property called browserAPIs which is a FOAM DAO. It contains interface and
+    API information for major browsers and version for recent two years. It
+    has methods to transform the data model from Google Cloud Datastore
+    to CSV format String or JSON structure to be displayed in client side.
+    It also has methods to do data analysis.`,
+  requires: [
+    'org.chromium.apis.web.BrowserAPI',
+    'foam.dao.EasyDAO',
+    'foam.dao.ArrayDAO',
+    'foam.mlang.ExpressionsSingleton',
+    'foam.mlang.sink.GroupBy',
+    'foam.mlang.sink.Count',
+  ],
+  properties: [
+    {
+      name: 'browserAPIs',
+      documentation: `This should be a server GCD DAO that contains
+        interfaece, API information for major browsers and versions
+        for recent 2 years. It is implemented as a MDAO for now.`,
+      factory: function() {
+        return this.EasyDAO.create({
+          name: 'BrowserAPIDAO',
+          of: this.BrowserAPI,
+          daoType: 'MDAO',
+        }).addPropertyIndex(this.BrowserAPI.BROWSER_KEY)
+        .addPropertyIndex(this.BrowserAPI.INTERFACE_KEY);
+      },
+    },
+    {
+      name: 'mlang',
+      documentation: `FOAM's mlang expressiongs, it supports Math and
```

Comment:
No need to say it's a FOAM mlang; clear from the context. However, please document the _why_ and the _how_: What's this used for? Is it usually injected or set by some other code later?